lambdas/service/app.py

```diff
@@ -1538,22 +1562,27 @@ def handle_manifest_generation_request() -> Tuple[int, Manifest]:
                                                            catalog=catalog,
                                                            filters=filters)
         if manifest is not None:
-            return 0, manifest
+            return amend_manifest_location(manifest=manifest)
     name = config.state_machine_name(generate_manifest.lambda_name)
     async_service = AsyncManifestService(name)
     try:
-        return async_service.start_or_inspect_manifest_generation(app.self_url(),
-                                                                  format_=format_,
-                                                                  catalog=catalog,
-                                                                  filters=filters,
-                                                                  token=token,
-                                                                  object_key=object_key)
+        time, manifest = async_service.start_or_inspect_manifest_generation(app.self_url(),
+                                                                            format_=format_,
+                                                                            catalog=catalog,
+                                                                            filters=filters,
+                                                                            token=token,
+                                                                            object_key=object_key)
     except ClientError as e:
         if e.response['Error']['Code'] == 'ExecutionDoesNotExist':
             raise BadRequestError('Invalid token given')
         raise
     except ValueError as e:
         raise BadRequestError(e.args)
+    else:
+        if time == 0:
+            return amend_manifest_location(manifest=manifest, token=token)
+        else:
+            return time, manifest
```

Comment:
Again, I don't see any benefit to writing out the keyword args here.
```suggestion
        if time == 0:
            manifest = amend_manifest_location(manifest, token)
        return time, manifest
```